### PR TITLE
Update webapi/sessions route

### PR DESCRIPTION
**Pull request by avatus**
Update the webapi routes to include `/web` to better clarify they are for web sessions only
webapps buddy: https://github.com/gravitational/webapps/pull/1486

### CODE REVIEW

File: lib/web/apiserver.go [hunk 1]
```diff
@@ -505,12 +505,13 @@ func (h *Handler) bindDefaultEndpoints() {
...
-	// Web sessions
-	h.POST("/webapi/sessions/web", httplib.WithCSRFProtection(h.WithLimiterHandlerFunc(h.createWebSession)))
+	// App sessions
 	h.POST("/webapi/sessions/app", h.WithAuth(h.createAppSession))
-	h.DELETE("/webapi/sessions", h.WithAuth(h.deleteSession))
-	h.POST("/webapi/sessions/renew", h.WithAuth(h.renewSession))
 
+	// Web sessions
+	h.POST("/webapi/sessions/web", httplib.WithCSRFProtection(h.WithLimiterHandlerFunc(h.createWebSession)))
+	h.DELETE("/webapi/sessions/web", h.WithAuth(h.deleteSession))
```
Comment (ibeckermayer) [inline lib/web/apiserver.go:513]: nit: let's call these `deleteWebSession` and `renewWebSession` for clarity/consistency.

**Review by ibeckermayer — APPROVED** [review]

**Review by kimlisa — APPROVED** [review]

File: lib/web/apiserver.go [hunk 2]
```diff
@@ -505,12 +505,13 @@ func (h *Handler) bindDefaultEndpoints() {
 	// Forwards traces to the configured upstream collector
 	h.POST("/webapi/traces", h.WithAuth(h.traces))
 
-	// Web sessions
-	h.POST("/webapi/sessions/web", httplib.WithCSRFProtection(h.WithLimiterHandlerFunc(h.createWebSession)))
+	// App sessions
 	h.POST("/webapi/sessions/app", h.WithAuth(h.createAppSession))
-	h.DELETE("/webapi/sessions", h.WithAuth(h.deleteSession))
```
Comment (codingllama) [inline lib/web/apiserver.go:511]: I wonder if we should keep the old routes around for a while to avoid breakages. Are these only ever used by the Web UI?

**Review by sfreiberg — APPROVED** [review]

File: lib/web/apiserver.go (hunk 2 above)
Comment (ibeckermayer) [reply lib/web/apiserver.go:511]: I only see them being used by the Web UI. Although maybe there's some compatibility guarantee we make towards 3rd party clients?

Comment (avatus) [reply lib/web/apiserver.go:511]: I'm ok with adding a "DELETE IN 12.0" as long as we don't finally get to deleting in 17.1 😂

Comment (avatus) [reply lib/web/apiserver.go:511]: Added a deprecated session for the older routes. Didn't know when to put a "DELETE" but left a comment to this PR to give context.

File: lib/web/apiserver.go [hunk 3]
```diff
@@ -505,12 +505,19 @@ func (h *Handler) bindDefaultEndpoints() {
...
-	h.POST("/webapi/sessions/renew", h.WithAuth(h.renewSession))
 
+	// Deprecated web sessions routes
```
Comment (codingllama) [inline lib/web/apiserver.go:511]: Suggestion:

```suggestion
	// DELETE IN 13, deprecated/unused web sessions routes (avatus)
```

We do grep for those (at least I do), so it's a useful mark. Username added for context, although it also makes it easy to find your own cleanups ;)

File: lib/web/apiserver.go (hunk 2 above)
Comment (codingllama) [reply lib/web/apiserver.go:511]: I think the compatibility guarantee is more in terms of binary versions than APIs, but this is more to play it safe.

**Review by codingllama — APPROVED** [review]

File: lib/web/apiserver.go (hunk 2 above)
Comment (zmb3) [reply lib/web/apiserver.go:511]: Let's put `DELETE IN 13` and let's go back and do this as soon as we cut the v12 branch next week so we don't forget again.

**Comment by avatus** [comment]
This PR was backported in the following PRs here
[v10](https://github.com/gravitational/teleport/pull/19870)
[v11](https://github.com/gravitational/teleport/pull/19869)

Thank you @sfreiberg !!

**Pull request merged**